governance/xc_admin/packages/xc_admin_common/src/price_store.ts

```diff
@@ -0,0 +1,236 @@
+          isWritable: true,
+        },
+        {
+          pubkey: configKey,
+          isSigner: false,
+          isWritable: true,
+        },
+        {
+          pubkey: SystemProgram.programId,
+          isSigner: false,
+          isWritable: false,
+        },
+      ],
+      programId: PRICE_STORE_PROGRAM_ID,
+      data: instructionData,
+    });
+  } else if (data.type == "InitializePublisher") {
+    const [configKey, configBump] = PublicKey.findProgramAddressSync(
+      [Buffer.from("CONFIG")],
+      PRICE_STORE_PROGRAM_ID
+    );
+    const [publisherConfigKey, publisherConfigBump] =
+      PublicKey.findProgramAddressSync(
+        [Buffer.from("PUBLISHER_CONFIG"), data.data.publisherKey.toBuffer()],
+        PRICE_STORE_PROGRAM_ID
+      );
+    const instructionData = Buffer.concat([
+      Buffer.from([
+        InstructionId.InitializePublisher,
+        configBump,
+        publisherConfigBump,
+      ]),
+      data.data.publisherKey.toBuffer(),
+    ]);
+    return new TransactionInstruction({
+      keys: [
+        {
+          pubkey: data.data.authorityKey,
+          isSigner: true,
+          isWritable: true,
+        },
+        {
+          pubkey: configKey,
+          isSigner: false,
+          isWritable: false,
+        },
+        {
+          pubkey: publisherConfigKey,
+          isSigner: false,
+          isWritable: true,
+        },
+        {
+          pubkey: data.data.bufferKey,
+          isSigner: false,
+          isWritable: true,
+        },
+        {
+          pubkey: SystemProgram.programId,
+          isSigner: false,
+          isWritable: false,
+        },
+      ],
+      programId: PRICE_STORE_PROGRAM_ID,
+      data: instructionData,
+    });
+  }
+  // No need to support SubmitPrices instruction.
+  throw new Error("invalid type");
+}
+
+export function parsePriceStoreInstruction(
+  instruction: TransactionInstruction
+): PriceStoreInstruction {
+  if (instruction.programId != PRICE_STORE_PROGRAM_ID) {
+    throw new Error("program ID mismatch");
+  }
+  if (instruction.data.length < 1) {
+    throw new Error("instruction data is too short");
+  }
+  const instructionId = instruction.data.readInt8(0);
+  if (instructionId == InstructionId.Initialize) {
+    if (instruction.data.length < 34) {
+      throw new Error("instruction data is too short");
+    }
+    const authorityKey = new PublicKey(instruction.data.subarray(2, 34));
+    if (instruction.keys.length != 3) {
+      throw new Error("invalid number of accounts");
+    }
+    return {
+      type: "Initialize",
+      data: {
+        payerKey: instruction.keys[0].pubkey,
+        authorityKey,
+      },
+    };
+  } else if (instructionId == InstructionId.InitializePublisher) {
+    if (instruction.data.length < 35) {
+      throw new Error("instruction data is too short");
+    }
+    const publisherKey = new PublicKey(instruction.data.subarray(3, 35));
+    if (instruction.keys.length != 5) {
```

Comment:
de we need to check remaining accounts to be valid?

Reply:
I've added a check that accounts and instruction data match the expected values.